tests/ui/binding/order-drop-with-match.rs

```diff
@@ -14,6 +14,7 @@ impl Drop for A {
     fn drop(&mut self) {
         unsafe {
             ORDER[INDEX] = 1;
+            //~^ WARN creating a reference to mutable static is discouraged [static_mut_refs]
```

Comment:
Since this test is unrelated, let's fix (or `allow(..)` and mark `FIXME`) this and other similar uses instead.